codalab/migration.py

```diff
@@ -0,0 +1,216 @@
+            uuids=[bundle_uuid], metadata_key={'store'}
+        )
+        assert metadata.get('store', None) is None
+
+        # storage_type is a legacy field, will still update this field because there is no side effect
+        self.bundle_manager._model.update_bundle(
+            bundle,
+            {
+                'storage_type': self.target_store_type.value,
+                'is_dir': is_dir,
+                'metadata': {'store': self.target_store_name},
+            },
+        )
+
+    def delete_origin_bundle(self, bundle_uuid, bundle_location):
+        # Delete data from orginal bundle store
+        if os.path.exists(bundle_location):
+            path_util.remove(bundle_location)
+
+
+if __name__ == '__main__':
+    # Command line parser, parse the worksheet id
+    parser = argparse.ArgumentParser(
+        description='Manages your local CodaLab Worksheets service deployment'
+    )
+    parser.add_argument('worksheet', type=str, help='The worksheet uuid that needs migration')
+    parser.add_argument('--target_store_name', type=str, help='The destination bundle store name')
+    parser.add_argument(
+        '-d',
+        '--dry-run',
+        help='Only upload the bundle to Azure, does not modify database',
+        action='store_true',
+    )
+    parser.add_argument(
+        '-k', '--keep', help='Keep bundle content in origin bundle store', action='store_true'
+    )
+
+    args = parser.parse_args()
+
+    worksheet_uuid = args.worksheet
+    target_store_name = (
+        "azure-store-default" if args.target_store_name is None else args.target_store_name
+    )
+    if not spec_util.UUID_REGEX.match(worksheet_uuid):
+        raise Exception("Input worksheet uuid has wrong format. ")
+
+    # TODO: write output to log / log files
+    migration = Migration(target_store_name)
+    migration.setUp()
+
+    bundle_uuids = migration.get_bundle_uuids(worksheet_uuid)
+
+    for bundle_uuid in bundle_uuids:
+        bundle = migration.get_bundle(bundle_uuid)
+        if bundle.bundle_type != 'dataset' or bundle.state != 'ready':
+            # only migrate uploaded bundle, and the bundle state needs to be ready
+            continue
+
+        # Uploaded bundles does not need has dependencies
+        assert len(bundle.dependencies) == 0
+
+        if migration.is_linked_bundle(bundle_uuid):
+            # Do not migrate link bundle
+            continue
+
+        bundle_location = migration.get_bundle_location(bundle_uuid)
+
+        if parse_linked_bundle_url(bundle_location).uses_beam:
+            # Do not migrate Azure / GCP bundles
+            continue
+
+        bundle_info = migration.get_bundle_info(bundle_uuid, bundle_location)
+
+        is_dir = bundle_info['type'] == 'directory'
+        migration.upload_to_Azure(bundle_uuid, bundle_location, is_dir)
+        migration.modify_bundle_data(bundle, bundle_uuid, is_dir)
+        migration.delete_origin_bundle(bundle_uuid, bundle_location)
```

Comment:
Before we run delete_origin_bundle, should we first do a sanity check? such as listing the files in the Azure version, listing the files in the disk version, and comparing them?